source/common/conn_pool/conn_pool_base.h

```diff
@@ -122,6 +122,15 @@ class ConnPoolImplBase : protected Logger::Loggable<Logger::Id::pool> {
     return *static_cast<T*>(&context);
   }
 
+  // Determines if prefetching is warranted based on the number of streams in
+  // use, pending streams, anticipated capacity, and preconnect configuration.
+  //
+  // If anticipate_incoming_stream is true this assumes a call to newStream is
+  // pending, which is true for global preconnect.
+  static bool shouldConnect(size_t pending_streams, size_t active_streams,
+                            uint32_t connecting_capacity, float preconnect_ratio,
```

Comment:
Pedantically for this to make sense this should be `connecting_and_connected_capacity`, right? Since it's not just about active streams, it's about all excess capacity and all pending or existing connections? If that is correct maybe update so it's more clear?